app/src/main/java/org/mozilla/focus/home/HomeFragment.kt

```diff
@@ -90,6 +92,20 @@ class HomeFragment : Fragment() {
         }
 
         registerForContextMenu(view)
+
+        tileContainer.addOnScrollListener(object : RecyclerView.OnScrollListener() {
+            override fun onScrolled(recyclerView: RecyclerView?, dx: Int, dy: Int) {
+                val gridLayoutManager = recyclerView?.layoutManager as GridLayoutManager
+                val lastVisibleItem = gridLayoutManager.findLastCompletelyVisibleItemPosition()
+                // If the focused tile position is greater than the last fully visible item position,
+                // we must add a scroll offset to reveal the next row of tiles to prevent the URL bar from being focused.
```

Comment:
Nit: I'd just say this is to hint that there are more items by revealing the next row - I know that "fixing jump of focus to urlbar" is what this bug is for, but that's more of a confusing side effect.